desc/magnetic_fields.py

```diff
@@ -1803,6 +1810,299 @@ def from_surface(
         )
 
 
+class OmnigenousField(Optimizable, IOAble):
+    """Model field with perfect omnigenity.
+
+    Notes
+    -----
+    Doesn't conform to MagneticField API, as it only knows about :math:`|B|` in flux
+    coordinates, not vector B in lab coordinates.
+
+    """
+
+    _io_attrs_ = [
+        "_L_well",
+        "_M_well",
+        "_L_shift",
+        "_M_shift",
+        "_N_shift",
+        "_NFP",
+        "_helicity",
+        "_well_basis",
+        "_shift_basis",
+        "_B_lm",
+        "_x_lmn",
+    ]
+
+    def __init__(
+        self,
+        L_well=0,
+        M_well=2,
+        L_shift=0,
+        M_shift=0,
+        N_shift=0,
+        NFP=1,
+        helicity=(1, 0),
+        B_lm=None,
+        x_lmn=None,
+    ):
+        self._L_well = int(L_well)
+        self._M_well = int(M_well)
+        self._L_shift = int(L_shift)
+        self._M_shift = int(M_shift)
+        self._N_shift = int(N_shift)
+        self._NFP = int(NFP)
+        self.helicity = helicity
+        self._well_basis = ChebyshevPolynomial(L=self.L_well)
+        self._shift_basis = ChebyshevDoubleFourierBasis(
+            L=self.L_shift,
+            M=self.M_shift,
+            N=self.N_shift,
+            NFP=self.NFP,
+            sym="cos(t)",
+        )
+        if B_lm is None:
+            self._B_lm = np.concatenate(
+                (
+                    np.ones((self.M_well,)),
+                    np.zeros((self.L_well * self.M_well,)),
+                )
+            )
+        else:
+            assert len(B_lm) == (self.L_well + 1) * self.M_well
+            self._B_lm = B_lm
+        if x_lmn is None:
+            self._x_lmn = np.zeros(self.shift_basis.num_modes)
+        else:
+            assert len(x_lmn) == self.shift_basis.num_modes
+            self._x_lmn = x_lmn
+
+        # TODO: should we not allow some types of helicity?
+        warnif(
+            self.helicity != (0, self.NFP) and self.helicity[0] != 1,
+            UserWarning,
+            "Typical helicity (M,N) has M=1.",
+        )
+        warnif(
+            self.helicity != (1, 0) and self.helicity[1] != self.NFP,
+            UserWarning,
+            "Typical helicity (M,N) has N=NFP.",
```

Comment:
there can be negative helicity as welll, with (1,-NFP) or (-1,NFP)

Reply:
I think this is fixed now